features/lorawan/lorastack/mac/LoRaMacCrypto.h

```diff
@@ -26,97 +26,150 @@ SPDX-License-Identifier: BSD-3-Clause
 #ifndef MBED_LORAWAN_MAC_LORAMAC_CRYPTO_H__
 #define MBED_LORAWAN_MAC_LORAMAC_CRYPTO_H__
 
-/**
- * Computes the LoRaMAC frame MIC field
- *
- * @param [in]  buffer          - Data buffer
- * @param [in]  size            - Data buffer size
- * @param [in]  key             - AES key to be used
- * @param [in]  address         - Frame address
- * @param [in]  dir             - Frame direction [0: uplink, 1: downlink]
- * @param [in]  seq_counter     - Frame sequence counter
- * @param [out] mic             - Computed MIC field
- *
- * @return                        0 if successful, or a cipher specific error code
- */
-int compute_mic(const uint8_t *buffer, uint16_t size, const uint8_t *key,
-                      uint32_t address, uint8_t dir, uint32_t seq_counter,
-                      uint32_t *mic);
+#include "mbedtls/aes.h"
+#include "mbedtls/cmac.h"
 
 /**
- * Performs payload encryption
- *
- * @param [in]  buffer          - Data buffer
- * @param [in]  size            - Data buffer size
- * @param [in]  key             - AES key to be used
- * @param [in]  address         - Frame address
- * @param [in]  dir             - Frame direction [0: uplink, 1: downlink]
- * @param [in]  seq_counter     - Frame sequence counter
- * @param [out] enc_buffer      - Encrypted buffer
- *
- * @return                        0 if successful, or a cipher specific error code
+ * CMAC/AES Message Integrity Code (MIC) Block B0 size
  */
-int encrypt_payload(const uint8_t *buffer, uint16_t size, const uint8_t *key,
-                    uint32_t address, uint8_t dir, uint32_t seq_counter,
-                    uint8_t *enc_buffer);
+#define LORAMAC_MIC_BLOCK_B0_SIZE       16
```

Comment:
Not a fan of this define - it's uniquely specified in one place - `sizeof mic_block_b0`. You can just use that to reference, and the size of the mic_block is obviously where it's defined: `mic_block_b0[16]`.

(I often try to avoid excessive defines, having been traumatised by a source file that had 100 lines like this:

    HW_REG_A = REG_A_INIT_VALUE;
    HW_REG_B = REG_B_INIT_VALUE;

Yay, pointless indirection. Don't need a define for a constant that occurs in exactly 1 place, unless you need to expose it as a config option.)